File: proxy/tests/script_engine/url_hostname.js

/// <reference path="../../script_libs/js/index.d.ts" />
/** @type {Extension} */
const url_hostname = {
  request(flow) {
    if (flow.request.url.hostname = "localhost") {
      flow.request.url.hostname = "example.com"
    }
  },
};
globalThis.extensions = [url_hostname];

File: proxy/tests/script_engine/url_hostname.lua
pcall(require, "../../script_libs/lua/roxy.lua")
---@type Extension
local url_hostname = {
	request = function(flow)
		if flow.request.url.hostname == "localhost" then
			flow.request.url.hostname = "example.com"
		end
	end,
}
Extensions = { url_hostname }
